doctor/scutil/scutil_darwin.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package scutil
+
+import (
+	"context"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"tailscale.com/types/logger"
+)
+
+func (Check) Run(ctx context.Context, logf logger.Logf) error {
+	cmd := exec.CommandContext(ctx, "scutil", "--dns")
```

Comment:
Somewhat surprised this works from the macOS sandbox (but it does). However, we won't be able to use it on iOS. 

Though more work (as far as having to rely on CGo), you could use the dnsinfo private API (see tailscale/corp#7680) which lets you get at this information more directly (it's what scutil uses in its implementation -- https://github.com/apple-oss-distributions/configd/blob/ebd2cef166d947659a001b076ea919351f7baf5d/scutil.tproj/tests.c#L703-L716).

Reply:
Would you prefer that I update this to rely on CGo, or are you okay to merge this as-is and follow up with the dnsinfo private API? I'm a bit leery of relying on that, though given that this is debug-only and how long it appears to have been around for it's probably fine?